### PR TITLE
#144 Support for dry-run/generate-only mode

### PR DESCRIPTION
**Pull request by csakshaug**
## Description
#144 Support for dry-run/generate-only mode

## How Has This Been Tested?
I executed and testet on multiple machines. After executed the dry run I can apply the files easy with the scripts.

## Checklist:

I have:

- [x] checked my changes follow the style of the existing code / OpenFaaS repos
- [x] updated the documentation and/or roadmap in README.md
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -172,7 +174,49 @@ func filesExists(files []types.FileSecret) error {
 	return nil
 }
 
+func dryProcess(plan types.Plan, prefs InstallPreferences) error {
```
Comment (alexellis) [inline main.go]: This seems to duplicate the work in the process method?

Comment (alexellis) [reply main.go]: @csakshaug thoughts?

Comment (csakshaug) [reply main.go]: Will give you feedback later today

Comment (csakshaug) [reply main.go]: We could either do a check inside the process for every "task/operation" or have their process where we take out the "task/operation" that make sense to execute in a dry-run. 

Instead call it dry-run perhaps we could focus on just updated the configurations, today the implementation update the yaml files and other scripts for ingress, tls and stack and execute a kubectl apply with dry-run feature.

I have uses this feature a lot for testing and updating the cluster with going through all the files or using the kubectl edit or reset+bootstrap again.

**Pull request closed without merging**